pkg/api/config_test.go

```diff
@@ -0,0 +1,336 @@
+package api_test
```

Comment:
`package api_test` is also not wrong. this is best practice for testing to avoid cyclic dependencies and in fact many of the standard library tests use this pattern. e.g.https://golang.org/src/math/all_test.go